Update pints.plot

chonlei opened the pull request:
Update pints.plot and should 
Fix #249 
Fix #263 
Fix #267

chonlei commented:
Need to change and update examples before merge.

chonlei commented on pints/plot.py:
```diff
@@ -168,65 +168,81 @@ def function_between_points(f, point_1, point_2, padding=0.25, evaluations=20):
     bins = 40
     alpha = 0.5
-    n_sample, n_param = chain.shape
+    samples = np.array(samples)  # try to handle numpy array only
```
Why we want it to be different sizes? (We cannot have `samples.shape()` next line then, if we are handling lists.)

MichaelClerx replied:
Yeah you can't have the shape line, but apart from that...

chonlei replied:
Fixed.

chonlei commented:
Ready to merge. Rebased onto master and updated all examples that use `pints.plot` including the new ones. This PR should

Close #273 
Close #270 
Close #267 
Close #263 
Close #249

MichaelClerx commented on pints/plot.py:
```diff
@@ -168,65 +168,105 @@ def function_between_points(f, point_1, point_2, padding=0.25, evaluations=20):
+                                     50 + n_percentiles / 2.)
+            xbins = np.linspace(xmin, xmax, bins)
+            axes[i].hist(samples_j[:, i], bins=xbins, alpha=alpha,
+                         label='Samples ' + str(1 + j_list))
```
Maybe something like "List 1", "List 2", etc.?

MichaelClerx commented:
Looks good @chonlei !
Can you maybe "Squash and merge" so that we have fewer commits? (If you do, make sure to write "Closes #1234 etc. in the new commit message you make).
